event.go

```diff
@@ -0,0 +1,164 @@
+package lookout
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"gopkg.in/sourcegraph/go-vcsurl.v1"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+type EventID [20]byte
+
+// ComputeEventID compute the hash for a given provider and content.
+func ComputeEventID(provider, content string) EventID {
+	var id EventID
+	h := sha1.New()
+	h.Write([]byte(provider))
+	h.Write([]byte("|"))
+	h.Write([]byte(content))
+	copy(id[:], h.Sum(nil))
+	return id
+}
+
+func (h EventID) IsZero() bool {
+	var empty EventID
+	return h == empty
+}
+
+func (h EventID) String() string {
+	return hex.EncodeToString(h[:])
+}
+
+type EventType int
+
+const (
+	_ EventType = iota
+	PushEventType
+	PullRequestEventType
+)
+
+// Event represents a repository event.
+type Event interface {
+	// ID returns the EventID.
+	ID() EventID
+	// Type returns the EventType, in order to identify the concreate type of
+	// the event.
+	Type() EventType
+	// Revision returns a commit revision, containing the head and the base of
+	// the changes.
+	Revision() *CommitRevision
+}
+
+// CommitRevision defines a range of commits, from a base to a head.
+type CommitRevision struct {
+	// Base of the revision.
+	Base ReferencePointer
```

Comment:
Once we merge this, we'll need to check if/how CommitRevision/ReferencePointer interact with structs in the analyzer/changes API. Maybe we can reuse some structs, or make them comply with a common interface.

Reply:
Yep, I am not sure if this is the best struct, but the concept is important.